osu.Server.DifficultyCalculator/Dockerfile

```diff
@@ -1,4 +1,22 @@
-FROM mcr.microsoft.com/dotnet/runtime:5.0
-COPY bin/Release/net5.0/publish/ App/
-WORKDIR /App
+FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build-env
+WORKDIR /app
+
+# Handle project files and dependencies first for caching benefits
+COPY *.sln ./
+COPY osu.Server.DifficultyCalculator/osu.Server.DifficultyCalculator.csproj ./osu.Server.DifficultyCalculator/
+COPY osu.Server.Queues.BeatmapProcessor/osu.Server.Queues.BeatmapProcessor.csproj ./osu.Server.Queues.BeatmapProcessor/
+
+RUN dotnet restore
+
+# Copy everything else and build
+COPY . ./
+WORKDIR /app/osu.Server.DifficultyCalculator
+RUN dotnet publish -c Release -o out
+# get rid of bloat
+RUN rm -rf ./out/runtimes ./out/osu.Game.Resources.dll ./out/osuTK.dll
```

Comment:
Are you sure that osuTK is bloat? It provides the implementation for `Vector2` which is used in hitobjects.

Reply:
Yeah, I also wasn't sure about `osuTK`. May actually cause issues.

Also `osuTK` shouldn't be that big?

Reply:
To keep things working for now I've remove the osuTK delete. Can revisit in the future (or just stop using osuTK in the future).